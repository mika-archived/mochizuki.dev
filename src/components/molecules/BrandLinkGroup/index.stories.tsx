import React from "react";
import { storiesOf } from "@storybook/react";

import BrandLinkGroup from ".";

storiesOf("molecules/BrandLinkGroup", module).add("default", () => (
  <>
    <BrandLinkGroup account="MikazukiFuyuno" label="Twitter" href="https://twitter.com/MikazukiFuyuno" icon="twitter" />
  </>
));
