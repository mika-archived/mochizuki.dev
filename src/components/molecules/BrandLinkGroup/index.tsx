import React from "react";
import styled from "styled-components";

import FontAwesome from "../../atoms/FontAwesome";
import Link from "../../atoms/Link";
import { Small } from "../../atoms/Typographies";

type Props = {
  account: string;
  label: string;
  href: string;
  icon: string;
};

const Account = styled(Small)`
  color: #888;
`;

const Anchor = styled(Link)`
  text-decoration: none;
`;

const FlexBox = styled.div`
  display: flex;
  flex-flow: row nowrap;
`;

const Label = styled(Small)`
  color: #333;
`;

const Icon = styled(FontAwesome)`
  font-size: 36px;
  line-height: 1.2;
  color: #333;
`;

const BrandLinkGroup: React.FC<Props> = ({ account, label, href, icon }) => {
  return (
    <Anchor href={href}>
      <FlexBox>
        <Icon prefix="brands" icon={icon} fixed />
        <div>
          <Label>{label}</Label>
          <Account>{account}</Account>
        </div>
      </FlexBox>
    </Anchor>
  );
};

export default BrandLinkGroup;
