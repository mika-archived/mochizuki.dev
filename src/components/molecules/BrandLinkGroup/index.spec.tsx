import "@testing-library/jest-dom/extend-expect";
import "jest-styled-components";

import React from "react";
import { cleanup, render } from "@testing-library/react";

import BrandLinkGroup from ".";

afterEach(cleanup);

test("does not have changes from snapshot", () => {
  const { asFragment } = render(<BrandLinkGroup account="MikazukiFuyuno" label="Twitter" href="https://twitter.com/MikazukiFuyuno" icon="twitter" />);

  expect(asFragment()).toMatchSnapshot();
});
