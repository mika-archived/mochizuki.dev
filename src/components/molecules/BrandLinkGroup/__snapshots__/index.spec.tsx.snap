// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`does not have changes from snapshot 1`] = `
<DocumentFragment>
  .c5 {
  margin: 0;
  font-family: "游ゴシック体","Yu Gothic",YuGothic,"ヒラギノ角ゴシック Pro","Hiragino Kaku Gothic Pro",メイリオ,Meiryo,sans-serif;
  font-size: 14px;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  overflow: hidden;
  color: #888;
  text-overflow: ellipsis;
}

.c0 {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row nowrap;
  -ms-flex-flow: row nowrap;
  flex-flow: row nowrap;
}

.c4 {
  margin: 0;
  font-family: "游ゴシック体","Yu Gothic",YuGothic,"ヒラギノ角ゴシック Pro","Hiragino Kaku Gothic Pro",メイリオ,Meiryo,sans-serif;
  font-size: 14px;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  overflow: hidden;
  color: #333;
  text-overflow: ellipsis;
}

.c2 {
  font-size: 36px;
  line-height: 1.2;
  color: #333;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: column nowrap;
  -ms-flex-flow: column nowrap;
  flex-flow: column nowrap;
  min-width: 0;
  margin-left: 6px;
}

<a
    class="c0"
    href="https://twitter.com/MikazukiFuyuno"
    rel="noreferrer noopener"
    target="_blank"
  >
    <div
      class="c1"
    >
      <i
        class="fab fa-twitter fa-fw c2"
      />
      <div
        class="c3"
      >
        <p
          class="sc-AxjAm sc-AxiKw c4"
        >
          Twitter
        </p>
        <p
          class="sc-AxjAm sc-AxiKw c5"
        >
          MikazukiFuyuno
        </p>
      </div>
    </div>
  </a>
</DocumentFragment>
`;
